CodeReview to 18 from 10

Fisher4537 opened the pull request:
Implemented Barcode Scanner with MLKit library.
Hello Alberto, this is a copy of the pull request that you can find here. For the code review it is not necessary that you review all the changes: it would be too much. You can ignore all the .xml file changes and review ONLY the .java files. Let me know if you have better proposal.

Fisher4537 committed:
added BarcodeExtractor to read barcode, bug on thread managment: critical lock held

Fisher4537 committed:
added a copy of missing xml from drawable-v24 to drawable and vice versa

Fisher4537 committed:
added the files mentioned in the last commit

Fisher4537 committed:
Merge branch 'CorrectResourcesBug' into BarcodeReader

Fisher4537 committed:
added constructor of BarcodeExtractor class

Fisher4537 committed:
bugfix barcodeView setted under OCRTextView

Fisher4537 committed:
edited landscaper layout of ResultActivity, added 'Barcode:' beforethe barcode number in ResultActivity

Fisher4537 committed:
added comments to simplify code reading

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/BarcodeExtractor.java:
```diff
@@ -0,0 +1,119 @@
+        // Firebase implementation tutorial at https://firebase.google.com/docs/ml-kit/android/read-barcodes
+        FirebaseVisionImage image = FirebaseVisionImage.fromBitmap(img);
+        FirebaseVisionBarcodeDetector detector = FirebaseVision.getInstance()
+                .getVisionBarcodeDetector();
```
Nice line-wrapping!

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/BarcodeExtractor.java:
```diff
@@ -0,0 +1,119 @@
+        //https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CountDownLatch.html
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        Task<List<FirebaseVisionBarcode>> task = detector.detectInImage(image) // this start an asynchronous thread
```
This line exceeds the column limit according to our Style document. May it be a simple solution to move the comment over the task declaration?

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/BarcodeExtractor.java:
```diff
@@ -0,0 +1,119 @@
+                                + getInfoFromBarcode(barcodes) + "\n" +
+                                "----- END OF THE RECOGNIZED TEXT -----");
+
+                        //analogous to signal
```
The comment was meant to say that, according to CountDownLatch documentation, the method countDown() down here works as a "signal()" method applied to a semaphore, wasn't it? May it result ambiguous or hard to understand?

Fisher4537 replied:
Since the link to the CountDownLatch documentation has been passed, would it be clearer by omitting "//analogous to signal" comment?
I think also "//decrement CountDownLatch by one" would be ok! What do you think about it?

AlbVal97 replied:
Yes, I think your last proposal should just be fine and less ambiguous. This way the reader could decide whether to read the documentation or not at his own risk

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/BarcodeExtractor.java:
```diff
@@ -0,0 +1,119 @@
+                    });*/
+
+        // if barcode is not detected yet
+        if(!task.isSuccessful()) {
```
I don't understand why the following try-catch statement doesn't follow braces rules according to our Style document. Was there a proper distinctive reason I haven't seen?

Fisher4537 replied:
I didn't pay attantion to the style because the proposed style document isn't approved yet! However the style is different from the rest of the code! I'm going to fix it! Thanks!

AlbVal97 replied:
Ok, so for the rest of the review I followed the not-yet-approved Style document down here:

[Documento di stile.docx](https://github.com/PrandiniUniPD/OCRCamera/files/2637127/Documento.di.stile.docx)

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java:
```diff
@@ -56,16 +73,18 @@ public void onClick(View view) {
-            mImageView.setImageBitmap(Bitmap.createScaledBitmap(lastPhoto, lastPhoto.getWidth(), lastPhoto.getHeight(), false));
+            mImageView.setImageBitmap(Bitmap.createScaledBitmap(lastPhoto, lastPhoto.getWidth(),
+                            lastPhoto.getHeight(),
+                            false));
```
Nice line-wrapping!

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java:
```diff
@@ -180,5 +213,79 @@ protected void onPreExecute() {
+
+        /**
+         * Costructor initialize:
+         * @param barcodeTextView the View responsable for barcode rapresentation
```
I'm not sure about "responsable" and "rapresentation". May it be that some vowels have been scrambled?

Fisher4537 replied:
Do you mean that the words aren't correct? What do you suggest?

AlbVal97 replied:
Exactly, I would replace "responsable" with "responsible" and "rapresentation" with "representation"

AlbVal97 commented on app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java:
```diff
@@ -180,5 +213,79 @@ protected void onPreExecute() {
+
+            if(lastPhoto != null) {
+                rawBarcode = new BarcodeExtractor().getTextFromImg(lastPhoto); //see BarcodeExtractor class
+
```
I don't understand why the following if-else statement doesn't follow braces rules according to our Style document.

Fisher4537 replied:
I didn't pay attantion to the style because the proposed style document isn't approved yet! However the style is different from the rest of the code! I'm going to fix it! Thanks!

AlbVal97 reviewed (commented):
The code is well built and easy to read: great work! The only inaccuracies I may have found are about strict following of our Style document rules. Furthermore, a lot of comments contain little "bad English" mistakes I have chosen not to point out because rarely they compromise code readablity, but I equally suggest to edit to improve the overall quality.

Fisher4537 committed:
improved code style

Fisher4537 committed:
removed hard code

Fisher4537 commented:
Thank to you, I've fixed the document style bug and deleted useless comments. I'm going to commit the last comment modification where I remove also some hardcode and then if you agree we could close the pull request!

AlbVal97 reviewed (approved):
Ok, I think all has been solved
